apps/web/src/components/Menu/GlobalSettings/SettingsModal.tsx

```diff
@@ -118,6 +128,27 @@ const SettingsModal: React.FC<React.PropsWithChildren<InjectedModalProps>> = ({
     }
   }
 
+  const handleDiableNotifications = async () => {
+    try {
+      if (isSubscribed) await unsubscribe()
+      setAllowNotifications(false)
+      toast.toastSuccess(Events.Unsubscribed.title, Events.Unsubscribed.message?.())
+    } catch (error) {
+      const errMessage = errorBuilder(Events.UnsubscribeError, error)
+      toast.toastWarning(Events.UnsubscribeError.title, errMessage)
+    }
+  }
+
+  const handleEnableNotifications = async () => {
+    try {
+      setAllowNotifications(true)
+      toast.toastSuccess(Events.NotificationsEnabled.title, Events.NotificationsEnabled.message?.())
+    } catch (error) {
+      const errMessage = errorBuilder(Events.NotificationsEnabledError, error)
+      toast.toastWarning(Events.NotificationsEnabledError.title, errMessage)
+    }
+  }
+
```

Comment:
please create a custom hook to handle the logic like what we did to other global settings